### PR TITLE
update the processor status.

### PR DESCRIPTION
**Pull request by larry-aptos**
* This change updates the processor status to include latest transaction timestamp in the table.

### CODE REVIEW

File: rust/processor/src/worker.rs [hunk 1]
```diff
@@ -356,6 +356,19 @@ impl Worker {
...
+                .last()
+                .unwrap()
+                .transactions
+                .as_slice()
+                .last()
+                .unwrap()
+                .timestamp
+                .clone();
+            let last_transaction_timestamp = last_transaction_pb_timestamp.map(|t| {
+                chrono::NaiveDateTime::from_timestamp_opt(t.seconds, t.nanos as u32).unwrap()
```
Comment (rtso) [inline rust/processor/src/worker.rs:369]: Use `timestamp_to_unixtime`

Comment (bowenyang007) [reply rust/processor/src/worker.rs:369]: Yes please!

File: rust/processor/src/schema.rs [hunk 2]
```diff
@@ -1197,6 +1198,10 @@ diesel::table! {
 }
 
 diesel::joinable!(block_metadata_transactions -> transactions (version));
+diesel::joinable!(move_modules -> transactions (transaction_version));
+diesel::joinable!(move_resources -> transactions (transaction_version));
+diesel::joinable!(table_items -> transactions (transaction_version));
+diesel::joinable!(write_set_changes -> transactions (transaction_version));
```
Comment (bowenyang007) [inline rust/processor/src/schema.rs:1204]: can you remove these manually for now? We have a PR coming that removes it once for all.

**Review by bowenyang007 — APPROVED** [review]
A couple of comments to address. Otherwise looks good!

File: rust/processor/src/worker.rs (hunk 1 above)
Comment (larry-aptos) [reply rust/processor/src/worker.rs]: done.

File: rust/processor/src/schema.rs (hunk 2 above)
Comment (larry-aptos) [reply rust/processor/src/schema.rs]: done.

File: rust/processor/src/worker.rs (hunk 1 above)
Comment (larry-aptos) [reply rust/processor/src/worker.rs:369]: just realized that the destination of this variable is db, and the expected type is TIMESTAMP. If we did it here, we might need another conversion back to NativeDate type.

Comment (rtso) [reply rust/processor/src/worker.rs:369]: i guess it depends on how it's gonna be used. all the prom metrics use unixtime, so if we're comparing against prom metrics it makes more sense to use unixtime. maybe 2 columns - one for human readable timestamp and one for unixtime

Comment (bowenyang007) [reply rust/processor/src/worker.rs:369]: Please use this https://github.com/aptos-labs/aptos-indexer-processors/blob/main/rust/processor/src/utils/util.rs#L234-L245

**Review by bowenyang007 — COMMENTED** [review]
please reuse utils function

**Comment by rtso** [comment]
Use this commit to fix the lint https://github.com/aptos-labs/aptos-indexer-processors/pull/230/commits/ffcc60b1c2fbaa7ebf3abe4e7c477360d6f3d7b4

**Comment by bowenyang007** [comment]
That PR isn't ready though. Just fix lint here.

**Review by bowenyang007 — CHANGES_REQUESTED** [review]
see above

File: rust/processor/src/worker.rs (hunk 1 above)
Comment (larry-aptos) [reply rust/processor/src/worker.rs]: good call. updated.

File: rust/processor/src/worker.rs [hunk 3]
```diff
@@ -356,6 +356,16 @@ impl Worker {
...
+            let last_transaction_pb_timestamp = transactions_batches
+                .last()
+                .unwrap()
+                .transactions
+                .as_slice()
+                .last()
+                .unwrap()
+                .timestamp
+                .clone();
+
```
Comment (bowenyang007) [inline rust/processor/src/worker.rs:368]: uh can't you just reuse batch_end_txn_timestamp lol.

**Review by bowenyang007 — COMMENTED** [review]
almost there. fix lint and don't copy the same code lol 
image

File: rust/processor/src/worker.rs (hunk 3 above)
Comment (larry-aptos) [reply rust/processor/src/worker.rs]: done

File: rust/processor/src/worker.rs [hunk 4]
```diff
@@ -631,6 +631,7 @@ impl Worker {
                     .map(|t| timestamp_to_iso(&t))
                     .unwrap_or_default(),
                 end_txn_timestamp_iso = batch_end_txn_timestamp
+                    .clone()
```
Comment (rtso) [inline rust/processor/src/worker.rs:634]: is this clone necessary?

Comment (larry-aptos) [reply rust/processor/src/worker.rs]: hmm yea the previous one is required; removed now.

File: rust/processor/src/models/default_models/move_modules.rs [hunk 5]
```diff
@@ -51,14 +52,14 @@ impl MoveModule {
...
                 .unwrap_or_default(),
             address: standardize_address(&write_module.address.to_string()),
-            bytecode: parsed_data.as_ref().map(|d| d.bytecode.clone()),
-            exposed_functions: parsed_data.as_ref().map(|d| d.exposed_functions.clone()),
-            friends: parsed_data.as_ref().map(|d| d.friends.clone()),
-            structs: parsed_data.as_ref().map(|d| d.structs.clone()),
+            bytecode: parsed_data.clone().map(|d| d.bytecode.clone()),
+            exposed_functions: parsed_data.clone().map(|d| d.exposed_functions.clone()),
+            friends: parsed_data.clone().map(|d| d.friends.clone()),
+            structs: parsed_data.clone().map(|d| d.structs.clone()),
```
Comment (rtso) [inline rust/processor/src/models/default_models/move_modules.rs:62]: this clone is also unnecessary

Comment (larry-aptos) [reply rust/processor/src/models/default_models/move_modules.rs]: done.

**Review by rtso — APPROVED** [review]

**Pull request merged**